app/models/manageiq/providers/nuage/network_manager/event_catcher/stream.rb

```diff
@@ -42,4 +48,39 @@ def connection
     end
     @connection
   end
+
+  private
+
+  def current_batch
+    Array.new(@batch.size) { @batch.pop }
+  end
+
+  def sleep_poll_normal
+    sleep(self.class.parent.worker_settings[:poll])
+  end
+
+  #
+  # A simple thread `Thread.new { connection.run }` should normally be returned here.
+  # But doing so we measure a huge performance drop: events are fetched from AMQP immediately,
+  # but their processing is delayed for ~10min, which is horrible. It seems to be a problem
+  # with qpid somehow eating resources of the main thread. A quick workaround for now is to
+  # periodically pause qpid thread for a few seconds to let the main process access the resources.
+  #
+  def qpid_thread
+    t_interval = self.class.parent.worker_settings[:qpid_rest_interval]
+    t_sleep = self.class.parent.worker_settings[:qpid_rest_time]
```

Comment:
Maybe make these two arguments? You can then configure the interval and sleep for the qpid thread from outside. I think it would be more flexible and easier to test.